packages/dx-react-grid/src/plugins/table-header-row.jsx

```diff
@@ -51,22 +52,24 @@ export class TableHeaderRow extends React.PureComponent {
                 setColumnSorting, groupByColumn,
                 changeTableColumnWidths, changeDraftTableColumnWidths,
               }) => {
-                const { name: columnName } = params.tableColumn.column;
+                const { tableColumn: { column, draft } } = params;
+                const { name: columnName } = column;
                 const groupingSupported = grouping !== undefined &&
-                    grouping.length < columns.length - 1;
+                  grouping.length < columns.length - 1;
 
                 return (
-                  <HeaderCell
+                  <CellLayout
                     {...params}
-                    column={params.tableColumn.column}
+                    column={column}
+                    draft={draft}
                     getMessage={getMessage}
+                    cellComponent={cellComponent}
```

Comment:
Why the Layout component pass rest props to the Cell component? I mean that it is not quite clear why layout should receive the allowSorting property, for example. Maybe we should use the same technique from the GroupingPanel plugin?